### PR TITLE
sidebar form add atp trigger

### PR DESCRIPTION
**Pull request by johnnieskywalker**
# [Auto Take Profit form](https://app.shortcut.com/oazo-apps/story/5542/ui-auto-take-profit-set-up-form-parameters)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
image
  
## How to test 🧪
- Check if form matches figma designs (note final min and max for slider for various colllaterals is separate story)
- check if all acceptance criteria are met
- check if there are no unexpected errors

### CODE REVIEW

File: features/automation/optimization/autoTakeProfit/state/autoTakeProfitFormChange.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  state: AutoTakeProfitFormChange,
+  action: AutoTakeProfitFormChangeAction,
+): AutoTakeProfitFormChange {
+  switch (action.type) {
+    case 'execution-coll-ratio':
+      return { ...state, executionCollRatio: action.executionCollRatio }
+    case 'current-form':
+      return { ...state, currentForm: action.currentForm }
+    case 'close-type':
+      return { ...state, collateralActive: action.toCollateral }
```
Comment (piotrkonowrocki) [inline features/automation/optimization/autoTakeProfit/state/autoTakeProfitFormChange.ts:34]: Question, not request change.

I see that Stop-Loss has it done in similar way. Is that something that needs to be passed to smart contract in that way? Variable is named `close-type`, but doesn't accepts a type but boolean? At least I think it's boolean.

File: features/automation/optimization/autoTakeProfit/controls/AutoTakeProfitFormControl.tsx [hunk 2]
```diff
@@ -18,7 +27,23 @@ export function AutoTakeProfitFormControl({
   isAutoTakeProfitActive,
   vault,
 }: AutoTakeProfitFormControlProps) {
+  const { uiChanges } = useAppContext()
+  const [autoTakeProfitState] = useUIChanges<AutoTakeProfitFormChange>(AUTO_TAKE_PROFIT_FORM_CHANGE)
+
   const feature = AutomationFeatures.AUTO_TAKE_PROFIT
+  const closePickerConfig = {
```
[the remark this answers is not in the document]
Comment (piotrkonowrocki) [reply features/automation/optimization/autoTakeProfit/controls/AutoTakeProfitFormControl.tsx:34]: Okay, now after I scrolled down I see that you actually created `getAutoTakeProfitStatus` method, but did not used it here 😀

File: features/automation/optimization/common/controls/OptimizationFormControl.tsx [hunk 3]
```diff
@@ -119,12 +126,14 @@ export function OptimizationFormControl({
         txHelpers={txHelpers}
         vault={vault}
       />
-      <AutoTakeProfitFormControl
-        autoBuyTriggerData={autoBuyTriggerData}
-        constantMultipleTriggerData={constantMultipleTriggerData}
-        isAutoTakeProfitActive={isAutoTakeProfitActive}
-        vault={vault}
-      />
+      {autoTakeProfitEnabled && (
```
Comment (piotrkonowrocki) [inline features/automation/optimization/common/controls/OptimizationFormControl.tsx:129]: I think it's not needed here. Auto-Take Profit banner is disabled if this feature is off, I also made sure that dropdown won't display Auto-Take Profit without feature flag enabled.

**Review by piotrkonowrocki — CHANGES_REQUESTED** [review]
Finished checking out code, proceed to manual testing.

File: features/automation/optimization/autoTakeProfit/controls/AutoTakeProfitFormControl.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnnieskywalker) [reply features/automation/optimization/autoTakeProfit/controls/AutoTakeProfitFormControl.tsx:34]: :facepalm:

File: features/automation/optimization/autoTakeProfit/sidebars/SetupAutoTakeProfit.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
+// import { Vault } from 'blockchain/vaults'
```
[the remark this answers is not in the document]
Comment (johnnieskywalker) [reply features/automation/optimization/autoTakeProfit/sidebars/SetupAutoTakeProfit.tsx:1]: ah I thin it's leftover from before merging with dev with some place holders

File: features/automation/optimization/autoTakeProfit/state/autoTakeProfitFormChange.ts (hunk 1 above)
Comment (piotrkonowrocki) [reply features/automation/optimization/autoTakeProfit/state/autoTakeProfitFormChange.ts:34]: Resolved in DM.

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**